client-app/src/app/activity-involvement/activity-inv-form.component.ts

```diff
@@ -0,0 +1,127 @@
+import { ActivityInvFetchService } from '../_services/activity-involvement-fetch.service';
+import { ServicesComponent } from '.././services/services.component';
+
+import { Activity } from '../_models/activity';
+import { Involvement } from '../_models/involvement';
+import { MemberFetchService } from '../_services/member-fetch.service';
+import { MembersComponent } from '@app/members/members.component';
+
+
+@Component({
+    selector: 'app-activity-inv-form',
+    templateUrl: './activity-inv-form.component.html',
+    styleUrls: ['./activity-inv-form.component.css']
+})
+export class ActivityInvFormComponent implements OnInit{
+
+    constructor(
+        private activityService: ActivityFetchService,
+        private activityInvService: ActivityInvFetchService,
+        private memberService: MemberFetchService,
+        private router: Router,
+        @Inject(MAT_DIALOG_DATA) public data: any) { }
+
+    public involvements;
+    public involvement;
+    public members;
+    public involvementsByName: string[];
+    public forPrinting: string[];
+
+    ngOnInit() {
+
+        // Getting all involvements from attached data
+        //this.involvements = this.data.involvements;
+		
+        this.involvement = this.data.involvement;
+        this.members=this.loadMembers();
+		
+		this.viewMembers(this.involvement);
+		console.log(this.members);
+		
+		
+        
+		
+        //this.involvementsByName = [];
+
+    }
+	
+	
+
+    updateInvolvement(involvement: Involvement, memberID: string) {
+        this.involvementsByName = [];
+        console.log("Involvement ID: " + involvement.id + " Member ID: " + memberID);
+        
+
+        var dontAdd: boolean;
+        dontAdd = false;
+
+        // checks  if a member is already a part of an involvement
+        involvement.memberIDs.forEach(element => {
+            if (element === memberID) {
+                console.log("Already a member!");
+                dontAdd = true;
+            }
+        });
+
+        // adds member if it's new to the activity
+        if (dontAdd === false) {
+            involvement.memberIDs.push(memberID);
+           
+        }
+		this.viewMembers(this.involvement);
+        this.forPrinting = this.involvementsByName;
+
+
+        // prints out the list of members in an involvement
+        this.involvementsByName.forEach(element => {
+            console.log("Name: " + element);
+        })
+
+        
+
+    }
+	
+	
+	private viewMembers(involvement: Involvement){
+		this.involvementsByName = [];
+		if((this.involvement.memberIDs!=null) && (this.members !=null)){
+			for (var i = 0; i < this.involvement.memberIDs.length; i++) {
+                for (var j = 0; j < this.members.length; j++) {
+                    if (this.involvement.memberIDs[i] === this.members[j].id) {
+                        this.involvementsByName.push(this.members[j].firstName.concat(" ".concat(this.members[j].lastName)))                      
```

Comment:
Curious as to why you chose to use .concat here. Was it necessary to make something work or was it just a best practice thing?

Reply:
Just using it to put them together, was having trouble getting them to add up with +, it was doing unexpected stuff.